synapse/storage/databases/main/account_data.py

```diff
@@ -459,9 +459,6 @@ async def add_account_data_to_room(
         content_json = json_encoder.encode(content)
 
         async with self._account_data_id_gen.get_next() as next_id:
-            # no need to lock here as room_account_data has a unique constraint
-            # on (user_id, room_id, account_data_type) so simple_upsert will
-            # retry if there is a conflict.
```

Comment:
Table seems to have been added with this constraint in migration 27/account_data.

Reply:
Similarly for account_data below